spec/predicates/vuln.md

```diff
@@ -0,0 +1,170 @@
+
+The fields that make up this predicate type are:
+
+The `subject` contains whatever software artifacts are to be associated with this vulnerability report document.
+The `predicate` contains a JSON-encoded data with the following fields:
+
+**scanner** object
+
+> There are lots of container image scanners such as Trivy, Grype, Clair, etc.
+> This field describes which scanner is used while performing a container image scan,
+> as well as version information and which Vulnerability DB is used.
+
+**scanner.uri** string (ResourceURI), optional
+
+> > URI indicating the identity of the source of the scanner.
+
+**scanner.version** string (ResourceURI), optional
+
+> > The version of the scanner.
+
+**scanner.db.uri** string (ResourceURI), optional
+
+> > > URI indicating the identity of the source of the Vulnerability DB.
+
+**scanner.db.version** string, optional
+
+> > > The version of the Vulnerability DB.
+
+**scanner.db.lastUpdate, required** string (timestamp)
+
+> > > The timestamp of when the vulnerability DB was updated last time.
+
+**scanner.result** list
+
+> > The result contains a list of vulnerabilities.
+> > This is the most important part of this field because it'll store the scan result as a whole. So, people might want
+> > to use this field to take decisions based on them by making use of Policy Engines tooling whether allow or deny these images.
+
+**scanner.result.[*].vulnerability** object
+
+> > > The vulnerability object defines information about each one of the vulnerabilities found by the scanner.
+
+**scanner.result.[*].vulnerability.id** string
+
+> > > > This is the identifier of the vulnerability, e.g. [GHSA-fxph-q3j8-mv87](https://github.com/advisories/GHSA-fxph-q3j8-mv87) whose CVE id is [CVE-2017-5645](https://nvd.nist.gov/vuln/detail/CVE-2017-5645).
+
+**scanner.result.[*].vulnerability.severity** object
+
+> > > > The severity contains a list to describe the severity of a vulnerability using one or more quantitative scoring method.
+
+**scanner.result.[*].vulnerability.severity.type** string
+
+> > > > > The type describes the quantitative method used to calculate the associated.
+
+**scanner.result.[*].vulnerability.severity.score** string
+
+> > > > > This is a string representing the severity score based on the selected method.
+
+**scanner.result.[*].vulnerability.annotations** list
+
+> > > > > This is a list of key/value pairs where scanners can add additional custom information.
+
+**metadata.scanStartedOn, required** string (timestamp)
+
+> > The timestamp of when the scan started.
+
+**metadata.scanFinishedOn, required** string (timestamp)
+
+> > The timestamp of when the scan completed.
+
+## Example
+
+```jsonc
+{
+  "_type": "https://in-toto.io/Statement/v1",
+  "subject": [
+    {
+      ...
+    }
+  ],
+  // Predicate:
+  "predicateType": "https://in-toto.io/attestation/vulns/v0.1",
+  "predicate": {
+    "invocation": {
+      "parameters": [],
+      // [ "--format=json", "--skip-db-update" ]
```

Comment:
nit: I think these comments are actually supposed to be in the various parameter fields, right?